opal/corelib/runtime.js

```diff
@@ -1321,7 +1321,7 @@
       // A splatted array must be copied
       return value.slice();
     }
-    else if (value['$respond_to?']('to_a', true)) {
+    else if (typeof(value.$to_a) === 'function' && !value.$to_a.$$stub) {
```

Comment:
Because MRI calls it. There's a [test for this scenario](https://github.com/ruby/spec/blob/master/language/variables_spec.rb#L385-L392) in rubyspec and it fails [in your CI build](https://travis-ci.org/opal/opal/jobs/136962065#L316).

Actually `opal/mini` doesn't need it, but to make it compatible with `opal/full` there should be some wrapping `if` condition.